src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -58,6 +58,13 @@ public CommonCompiler(CommandLineParser parser, string responseFile, string[] ar
             this.Arguments = parser.Parse(allArgs, baseDirectory, sdkDirectoryOpt, additionalReferenceDirectories);
             this.MessageProvider = parser.MessageProvider;
             this.AnalyzerLoader = analyzerLoader;
+
+#if DEBUG
```

Comment:
Glad you agree.  This is a bit of a first (using features to implement a diagnostic switch) so I wanted to get feedback here before I settled on this approach.